UserTools/DirectionGridCheck/DirectionGridCheck.cpp

```diff
@@ -0,0 +1,245 @@
+#include "DirectionGridCheck.h"
+
+DirectionGridCheck::DirectionGridCheck():Tool(){}
+
+
+bool DirectionGridCheck::Initialise(std::string configfile, DataModel &data){
+
+  /////////////////// Useful header ///////////////////////
+  if(configfile!="") m_variables.Initialise(configfile); // loading config file
+  //m_variables.Print();
+
+  m_variables.Get("OutputFile", OutputFile);
+  m_variables.Get("ShowEvent", showEvent);
+  m_variables.Get("verbosity", verbosity);
+
+  fOutput_tfile = new TFile("DGCheck.root", "recreate");
+  AngularPlot = new TGraph();
+  AngularPlot->SetTitle("Seed Directions in Angular Space");
+  AngularPlot->GetXaxis()->SetTitle("Polar Angle");
+  AngularPlot->GetYaxis()->SetTitle("ZenithAngle");
+  AngularPlot->SetMarkerStyle(3);
+  AngularPlot->SetMarkerColor(1);
+  AngularPlot->SetLineColor(0);
+  TrueDirPlot = new TGraph();
+  TrueDirPlot->SetTitle("TrueDirection");
+  TrueDirPlot->SetMarkerColor(3);
+  TrueDirPlot->SetMarkerStyle(25);
+  TrueDirPlot->SetLineColor(0);
+  MRDDirPlot = new TGraph();
+  MRDDirPlot->SetTitle("MRDDirection");
+  MRDDirPlot->SetMarkerColor(2);
+  MRDDirPlot->SetMarkerStyle(26);
+  MRDDirPlot->SetLineColor(0);
+  SimpDirPlot = new TGraph();
+  SimpDirPlot->SetTitle("SimpleDirection");
+  SimpDirPlot->SetMarkerColor(4);
+  SimpDirPlot->SetMarkerStyle(27);
+  SimpDirPlot->SetLineColor(0);
+  RecoDirPlot = new TGraph();
+  RecoDirPlot->SetMarkerColor(5);
+  RecoDirPlot->SetMarkerStyle(28);
+  RecoDirPlot->SetLineColor(0);
+
+  m_data= &data; //assigning transient data pointer
+  /////////////////////////////////////////////////////////////////
+
+  return true;
+}
+
+
+bool DirectionGridCheck::Execute(){
+
+    auto* annie_event = m_data->Stores["RecoEvent"];
```

Comment:
I would advise against this method of check.
`m_data->Stores` is a `std::map`. The `std::map operator[]` will either return an element if it exists, or *create the element if it doesn't*. While it will create a default pointer that will _probably_ be null, I'm honestly not sure if it's guranteed to be, and either way this will result in `m_data->Stores` having a `RecoEvent` entry that is a nullptr. This is very effectively leaving a segfault landmine for someone else to step on.
```
if(m_data->Stores.count("RecoEvent")==0){
    Log("No RecoEvent Store!", v_error, verbosity);
    return false;
 }
BoostStore* reco_event = m_data->Stores.at("RecoEvent");
if(reco_event==nullptr){
    Log("RecoEvent Store is null!", v_error, verbosity);
 }
 ```
is more code, but it's safer and more explicit.

I would also advise against `auto*` - the use of `auto` is already unclear, but `auto*` is doubly obfuscated to novice coders.
Also, you're getting "RecoEvent" into `annie_event`, and your following error message prints `The PhaseITreeMaker tool could not find the ANNIEEvent Store!", which is both the wrong tool and wrong Store name.....